observability/src/main/java/io/grpc/observability/interceptors/GcpLogHelper.java

```diff
@@ -0,0 +1,430 @@
+        T message,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+      checkArgument(
+          eventType == EventType.GRPC_CALL_REQUEST_MESSAGE
+              || eventType == EventType.GRPC_CALL_RESPONSE_MESSAGE,
+          "event type must correspond to client message or server message");
+      checkNotNull(message, "message");
+
+      // TODO(dnvindhya): Convert message to bystestring
+      // byte[] messageArray = (byte[])message;
+      // int messageLength = messageArray.length;
+      // ByteString messageData =
+      // ByteString.copyFrom((byte[]) message, 0, ((byte[]) message).length);
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(eventType)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINE, "Writing RPC message to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+
+    /**
+     * Logs half close.
+     */
+    void logHalfClose(
+        long seqId,
+        String serviceName,
+        String methodName,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(EventType.GRPC_CALL_HALF_CLOSE)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINE, "Writing Half Close event to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+
+    /**
+     * Logs cancellation.
+     */
+    void logCancel(
+        long seqId,
+        String serviceName,
+        String methodName,
+        GrpcLogRecord.EventLogger eventLogger,
+        String rpcId) {
+      checkNotNull(serviceName, "serviceName");
+      checkNotNull(methodName, "methodName");
+      checkNotNull(rpcId, "rpcId");
+
+      GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+          .setSequenceId(seqId)
+          .setServiceName(serviceName)
+          .setMethodName(methodName)
+          .setEventType(EventType.GRPC_CALL_CANCEL)
+          .setEventLogger(eventLogger)
+          .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+          .setRpcId(rpcId);
+      logger.log(Level.FINE, "Writing Cancel event to GcpLogSink");
+      sink.write(logEntryBuilder.build());
+    }
+  }
+
+  // Copied from internal
+  static final class ByteArrayMarshaller implements Marshaller<byte[]> {
```

Comment:
Why is this class defined here? It is used only in tests of `InternalLoggingChannelInterceptorTest`. Is this going to be used in the main code in the future?

Reply:
For now, it is only used in tests. Once I start logging RPC messages I will move it its own file. Does that sound okay? 

Another option, we can also change access permissions of the class in Binary Logging based on what we decide and remove the definition from here.